### DISCUSSION TITLE
How can I maintain symmetrical edge_masks for edge(i, j) and (j, i) when explaining a node in an undirected graph by GNNexplainer?

### DISCUSSION THREAD

**Discussion by NBStarry** [pyg-team/pytorch_geometric#5964] [Q&A] [answered]
Hi, 
The question is as the title. Moreover, the subgraph may contains 100k+ edges. I haven't found any feasible solutions to maintain symmetrical edge_masks. The point is, I want to treat edge (i, j) and (j, i) as a whole when explaining an undirected graph, but the model requires them as input seperately. How to realize it?
Thanks in advance.

**Comment by rusty1s** [comment, marked as answer]
I don't think this is possible right now. `GNNExplainer` does not know about reverse edges since both end points will receive different messages. What about merging the scores from both directions in a post-processing step?

**Reply by NBStarry** [reply]
Thanks. By the way, when explaining node $i$, only edges whose targets are $i$ should be taken into consideration for most models in pyg. Is that right?

**Reply by rusty1s** [reply]
Yes, only local edges which are used during message passing will be considered for a single node.
